Use pulse configuration in fake Bogota, Rome, Manila and Santiago

omarcostahamido opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
-->

### Summary

this fixes #7685 

### Details and comments

jakelishman commented on qiskit/test/mock/backends/bogota/fake_bogota.py:
```diff
@@ -15,23 +15,24 @@
 
 
-class FakeLegacyBogota(fake_qasm_backend.FakeQasmLegacyBackend):
+class FakeLegacyBogota(fake_pulse_backend.FakePulseBackend):
```
This needs to be a type of legacy backend.

omarcostahamido replied:
Great catch. Thank you. I'm on to it.

jakelishman reviewed (commented):
Thanks for your interest in helping out on Qiskit!

If updating the one comment I've left is all that's required to fix the test failures, I'm happy to accept the PR.  If it needs more work and care, or if the definitions are too out-of-date, I'm not super keen to put in the reviewer time, though - change for change's sake isn't great for maintainability, and unless there's buggy behaviour, it's better for us to keep consistency in our public API.

If the tests pass after swapping `LegacyBogota` to be a `PulseLegacyBackend`, this will need a one-line "feature" release note saying that `FakeBogota`, etc now contain some sample pulse calibration data.

jakelishman commented:
@omarcostahamido: this has all gone through successfully - if you'd like to add a short release note, I'll get this merged.

omarcostahamido commented:
@jakelishman Thank you. I've added the release notes and updated the branch.

(feels good to not have spent time studying this and just produce 🗑️ ...)

mtreinish commented on releasenotes/notes/bogota-manila-rome-santiago-as-fakepulsebackends-2907dec149997a27.yaml:
```diff
@@ -0,0 +1,6 @@
+---
+prelude: >
+    FakeBogota, FakeManila, FakeRome, and FakeSantiago can now be used as fake backends in Pulse experiments.
```
You don't need a prelude section here. This section is reserved for the high level overview/highlights of the release. The contents of all the prelude sections are merged together into a single text section so we typically just write a signle release note with this section at release time.

```suggestion
```

omarcostahamido replied:
Ah ok. Thank you for clarifying that @mtreinish !

mtreinish commented on releasenotes/notes/bogota-manila-rome-santiago-as-fakepulsebackends-2907dec149997a27.yaml:
```diff
@@ -0,0 +1,6 @@
+    FakeBogota, FakeManila, FakeRome, and FakeSantiago can now be used as fake backends in Pulse experiments.
+features:
+  - |
+    FakeBogota, FakeManila, FakeRome, and FakeSantiago can now be used as fake backends in Pulse experiments.
```
```suggestion
    The ``FakeBogota``, ``FakeManila``, ``FakeRome``, and ``FakeSantiago`` fake backends which can be found
    in the ``qiskit.test.mock`` module can now be used as backends in Pulse experiments as they now include the
    a :class:`~qiskit.providers.models.PulseDefaults` created from a snapshot of the IBM Quantum's properties.
```

omarcostahamido replied:
Ok thank you so much for the added articulation.
Question: where can I learn about this notation you used here
>:class:`~qiskit.providers.models.PulseDefaults`

Meaning, how should I know that I need to write :class: and the "~" ?
@mtreinish

mtreinish reviewed (commented):
LGTM, just some small suggestions in the release note then I think this is ready to merge

omarcostahamido commented:
Ok thank you so much for the added articulation @mtreinish 
Question: where can I learn about this notation you used here
>:class:`~qiskit.providers.models.PulseDefaults`

Meaning, how should I know that I need to write :class: and the "~" ?

mtreinish commented:
It's the sphinx syntax for cross-references, which is documented here: https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-python-objects (we don't need the `:py` prefix here as it's all python)

omarcostahamido commented:
Thank you for the reference @mtreinish !

All checks have passed ✅ 

😄

jakelishman commented on releasenotes/notes/bogota-manila-rome-santiago-as-fakepulsebackends-2907dec149997a27.yaml:
```diff
@@ -0,0 +1,6 @@
+features:
+  - |
+    The ``FakeBogota``, ``FakeManila``, ``FakeRome``, and ``FakeSantiago`` fake backends which can be found
+    in the ``qiskit.test.mock`` module can now be used as backends in Pulse experiments as they now include the
```
```suggestion
    in the ``qiskit.test.mock`` module can now be used as backends in Pulse experiments as they now include
```

HuangJunye commented:
@omarcostahamido I am working on #7643 and found this problem. Thanks for fixing this!

omarcostahamido commented:
@HuangJunye my pleasure :)
Looking forward to see the work you're doing there when it gets merged as well.
It is funny that even fake backends have different generations now.
Perhaps, like the real QC hardware,  they could also have generation codenames: mockingbirdv1, mockingbirdv2, etc 😄 
I just created my first custom>merged notification on your PR (was about to ask how to do this), I might need to update my code then. 🙏
